BootloaderCorePkg/Library/MpInitLib/Ia32/MpFuncs.nasm

```diff
@@ -23,23 +23,32 @@ SECTION .text
 ;-------------------------------------------------------------------------------------
 ;RendezvousFunnelProc  procedure follows. All APs execute their procedure. This
 ;procedure does NOT serialize the AP processors. So this procedure cannot be the
-;vector for a boradcast IPI. 
-; 
+;vector for a boradcast IPI.
+;
 ;It must be noted that APs arrive here very raw...ie: real mode, no stack.
 ;-------------------------------------------------------------------------------------
 SPIN_LOCK_RELEASED   EQU   0
 SPIN_LOCK_ACQUIRED   EQU   1
 
- 
+
 global ASM_PFX(RendezvousFunnelProc)
 ASM_PFX(RendezvousFunnelProc):
 RendezvousFunnelProcStart:
 ; At this point CS = 0x(vv00) and ip= 0x0.
-  
+  db 0fh,  20h, 0d0h            ; mov        eax, cr2
+  db 66h,  09h, 0c0h            ; or         eax, eax
+
+  ; For SMM rebase flow, CR2 will be pre-filled with SMBASE
+  jz StartIpi
+  db 0eah                       ; jmp        0x3800:StartIpi
+  dw StartIpi - RendezvousFunnelProcStart
```

Comment:
Maybe a comment on why the jump is necessary. When CPU enters SMM, the CS will be initialized to 0x3000 and the code is written to use a base of 0x3800.

Reply:
Sure, will add that.